services/streammanager/kafka/client/producer.go

```diff
@@ -0,0 +1,120 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/segmentio/kafka-go"
+)
+
+type ProducerConfig struct {
```

Comment:
I believe we need another configuration field for the partitioner as well. Now, we are using `sarama.NewReferenceHashPartitioner` now, I assume this is the same with segment's `kafka.Murmur2Balancer`

Reply:
I think we should use `*kafka.Hash` to replace the `sarama.NewReferenceHashPartitioner` unless I'm misunderstanding something. That's why I hard-coded it in the producer [here](https://github.com/rudderlabs/rudder-server/blob/99fe96e42ccc97e9400000a585584d99dec999e7/services/streammanager/kafka/client/producer.go#L67). Why do you say we should use the murmur2 one, can you share the relevant docs?

Reply:
[here](https://github.com/Shopify/sarama/blob/main/partitioner.go#L170)
> NewReferenceHashPartitioner is like NewHashPartitioner except that it handles absolute values in the same way as the reference Java implementation. NewHashPartitioner was supposed to do that but it had a mistake and now there are people depending on both behaviors. This will all go away on the next major version bump.

[here](https://github.com/segmentio/kafka-go#java)
> Use the kafka.Murmur2Balancer balancer to get the same behaviour as the canonical Java client's default partitioner.

Reply:
@atzoum https://github.com/segmentio/kafka-go/pull/906

Reply:
I'm using my own fork for now till we hear from them.